pkgs/development/interpreters/luajit/default.nix

```diff
@@ -52,6 +52,11 @@ let
     ++ optional enableVMAssertions "-DLUAJIT_USE_ASSERT"
     ++ optional deterministicStringIds "-DLUAJIT_SECURITY_STRID=0"
   ;
+
+  buildStdenv = if buildPackages.stdenv.isx86_64 && stdenv.is32bit
+    then buildPackages.pkgsi686Linux.buildPackages.stdenv
+    else buildPackages.stdenv;
```

Comment:
While this looks good, I wonder if there is a more universal solution that works on other crosses.

Reply:
Honestly, I consider the luajit approach of requiring the bitness of the build host to match the build system pretty stupid. It is also the only package I know about that requires it (..although I have not looked for it. It is just from my practice with maintaining packages for OpenWrt.).

I am also afraid that this can't be easily achieved on other platforms. Only some CPUs have the ability to run 32-bit code alongside 64-bit. It is standard on x86\_64 but not necessary on ARM or RISC-V (I think that one of the examples is Apple's chips).

I should have added a comment there that it is a speciality of luajit, not some common approach.